File: domain/src/main/java/com/example/domain/seat/entity/Seat.java
package com.example.domain.seat.entity;

import com.example.domain.screening.entity.Screening;
import jakarta.persistence.*;
import lombok.Getter;

@Entity
@Table(name = "Seat")
@Getter
public class Seat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Integer seatNumber;

    @ManyToOne
    @JoinColumn(name = "screening_id", nullable = false)
    private Screening screening;
}

File: cinema/src/main/kotlin/com/redis/domain/Seat.kt
package com.redis.domain

import com.redis.domain.Cinema
import com.redis.utils.BaseEntity
import jakarta.persistence.*

@Table(name = "seat")
@Entity
class Seat(

    @Column(name = "seat_row")
    val row: String,

    @Column(name = "seat_col")
    val col: String,

    @ManyToOne
    @JoinColumn(name = "cinema_id")
    val cinema: Cinema,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seat_id")
    val id: Long? = null
): BaseEntity() {

}